Make `::-webkit-loading-auto-fill-button` an internal pseudo-element

nt1m opened the pull request:
#### f72d16287b1b22b75a471b272346ff74ab298fa4
<pre>
Make `::-webkit-loading-auto-fill-button` an internal pseudo-element
<a href="https://bugs.webkit.org/show_bug.cgi?id=267169">https://bugs.webkit.org/show_bug.cgi?id=267169</a>
<a href="https://rdar.apple.com/120579213">rdar://120579213</a>

Reviewed by Darin Adler.

This was added (relatively) recently in 252146@main in 2022 so it should be safe to unexpose.

Additional reasons the risk of compat fallout is low are:
- There are no hits on Github currently
- There are no hits on Apple&apos;s code search site as well
- ::-webkit- prefixed pseudo-elements have special parsing quirks for compatibility so they do not disable declarations if unrecognized
(see selectors specification: <a href="https://drafts.csswg.org/selectors-4/#compat)">https://drafts.csswg.org/selectors-4/#compat)</a>

* LayoutTests/fast/css/pseudo-class-internal-expected.txt: Removed.
* LayoutTests/fast/css/pseudo-class-internal.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/invalid-pseudos-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/invalid-pseudos.html: Added.

Add tests and remove obsolete one.

* Source/WebCore/css/CSSPseudoSelectors.json: Rename pseudo.

* Source/WebCore/css/html.css: Rename usages.
(input::-internal-loading-auto-fill-button):
(input::-webkit-loading-auto-fill-button): Deleted.

* Source/WebCore/css/process-css-pseudo-selectors.py:

Fix a bug in code generation where it was generating this invalid code:

if (!context.mode == UASheetMode &amp;&amp; ...)

Wrap the condition inside parentheses to prevent that.

* Source/WebCore/html/TextFieldInputType.cpp: Use new UserAgentPartId
(WebCore::autoFillButtonTypeToAutoFillButtonPseudoClassName):
(WebCore::isAutoFillButtonTypeChanged):

Canonical link: <a href="https://commits.webkit.org/272736@main">https://commits.webkit.org/272736@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3af186d56e688ac42547d94f9be0fe8157d205a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32866 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11623 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34739 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35487 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/29689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33801 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13963 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8799 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29141 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33319 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/9799 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/29354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/8535 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/8679 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/29310 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36820 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/29856 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/29717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/34813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/6764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32670 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/10491 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/28967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4238 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9424 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/fast/selectors/internal-pseudos.html:
```diff
@@ -0,0 +1,14 @@
+<title>Selectors: internal pseudos are not web exposed</title>
+<script src="../../resources/testharness.js"></script>
+<script src="../../resources/testharnessreport.js"></script>
+<script src="../../imported/w3c/web-platform-tests/css/support/parsing-testcommon.js"></script>
```
I’m a little worried about tying tests outside WPT to the structure of WPT. Maybe we could contribute this to WPT. No platform or engine would want these pseudo selectors to be exposed, whether they existed or not, so it would be OK for others.

nt1m replied:
That sounds reasonable.

nt1m committed:
Make `::-webkit-loading-auto-fill-button` an internal pseudo-element
https://bugs.webkit.org/show_bug.cgi?id=267169
rdar://120579213

Reviewed by Darin Adler.

This was added (relatively) recently in 252146@main in 2022 so it should be safe to unexpose.

Additional reasons the risk of compat fallout is low are:
- There are no hits on Github currently
- There are no hits on Apple's code search site as well
- ::-webkit- prefixed pseudo-elements have special parsing quirks for compatibility so they do not disable declarations if unrecognized
(see selectors specification: https://drafts.csswg.org/selectors-4/#compat)

* LayoutTests/fast/css/pseudo-class-internal-expected.txt: Removed.
* LayoutTests/fast/css/pseudo-class-internal.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/invalid-pseudos-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/invalid-pseudos.html: Added.

Add tests and remove obsolete one.

* Source/WebCore/css/CSSPseudoSelectors.json: Rename pseudo.

* Source/WebCore/css/html.css: Rename usages.
(input::-internal-loading-auto-fill-button):
(input::-webkit-loading-auto-fill-button): Deleted.

* Source/WebCore/css/process-css-pseudo-selectors.py:

Fix a bug in code generation where it was generating this invalid code:

if (!context.mode == UASheetMode && ...)

Wrap the condition inside parentheses to prevent that.

* Source/WebCore/html/TextFieldInputType.cpp: Use new UserAgentPartId
(WebCore::autoFillButtonTypeToAutoFillButtonPseudoClassName):
(WebCore::isAutoFillButtonTypeChanged):

Canonical link: https://commits.webkit.org/272736@main
